### PR TITLE
[Go] Jamessouth go constants 1919

### PR DESCRIPTION
**Pull request by jamessouth**
For your review:
#1919 
😃

### CODE REVIEW

File: languages/go/exercises/concept/constants/.docs/instructions.md
Comment (tehsphinx) [inline languages/go/exercises/concept/constants/.docs/instructions.md:16]: Aren't there 365 days in a year?

File: languages/go/exercises/concept/constants/.docs/after.md
Comment (tehsphinx) [inline languages/go/exercises/concept/constants/.docs/after.md:16]:
```suggestion
    const number2 = 2 // 2 is an untyped numeric constant and does not need to be explicitly given the type float64 as required by the Sqrt method
    sqrt2 := math.Sqrt(number2)
```

Comment (tehsphinx) [reply languages/go/exercises/concept/constants/.docs/after.md:16]: Wasn't really aware of this 👍. Learned something today!

Note that if you use math.Sqrt(2), 2 is a basic literal, not really a constant (AFAIK). The rules might be the same but I'd make it more obvious with the suggested change.

**Review by tehsphinx — COMMENTED** [review]
Very nice, @jamessouth!

I added 2 comments I think we should improve and then we can merge this! ❤️

**Comment by tehsphinx** [comment]
Note: sry for the long wait! Crazy time at work...

File: languages/go/exercises/concept/constants/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/introduction.md:2]: If there is just one section, I would suggest to remove the heading.

File: languages/go/exercises/concept/constants/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/introduction.md:6]:
```suggestion
const myWebsite = "Exercism"
```

File: languages/go/exercises/concept/constants/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/introduction.md:12]:
```suggestion
const myWebsite string = "Exercism"
```

File: languages/go/exercises/concept/constants/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/introduction.md:28]:
```suggestion
const (
    a = 9
    b
    c
    d = iota
    e
    f
    g
)
fmt.Print(a, b, c, d, e, f, g)
// Output: 9 9 9 3 4 5 6
```

File: languages/go/exercises/concept/constants/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/introduction.md:1]: I like how you've managed to keep the introduction short!

File: languages/go/exercises/concept/constants/.docs/instructions.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/instructions.md:1]: I can see how the first task fits in with the theme, but I feel thatthe other, date-related tasks make less sense. This might be due to the fact that the `constants` concept by itself is not _that_ interesting. Therefore, it might be convenient to combine this concept with another concept.

Comment (jamessouth) [reply languages/go/exercises/concept/constants/.docs/instructions.md:1]: Yeah this one was hard. Date-related tasks were used because dates are integral to the calculation of interest on loans and deposits. The `basics` concept is where variables are introduced and that would seem to be the natural destination for the `constants` concept if you don't want it to stand alone.

Comment (ErikSchierboom) [reply languages/go/exercises/concept/constants/.docs/instructions.md:1]: @jamessouth That might be best discussed with the other Go maintainers. Looking at the amount of text in the introduction.md document, I don't think it makes sense to add it to the basics exercise as that would become too big I feel (unless only a basic constants introduction was given). An alternative would be to combine this concept with another concept.

Comment (tehsphinx) [reply languages/go/exercises/concept/constants/.docs/instructions.md:1]: I think there is enough potential in constants to be a standalone exercise. Let us concentrate on merging this first, and then we can see how to improve it.

@jamessouth: could you address the open issues and then we can merge this?

File: languages/go/exercises/concept/constants/.docs/instructions.md
Comment (jamessouth) [reply languages/go/exercises/concept/constants/.docs/instructions.md:16]: Hi! I'm working on this now, sorry for the delay. 360 days is commonly used for interest calculation, at least in the USA.

File: languages/go/exercises/concept/constants/.docs/after.md
Comment (jamessouth) [reply languages/go/exercises/concept/constants/.docs/after.md:16]: That example is straight out of Rob Pike's `constants` [blog post](https://blog.golang.org/constants#TOC_12.). I will expand it like you have.

**Comment by jamessouth** [comment]
Changes made for your review 😀

**Review by oanaOM — APPROVED** [review]

**Comment by oanaOM** [comment]
@jamessouth great job! I think this is a solid exercise for now and we can merge it. @tehsphinx thoughts?

File: languages/go/exercises/concept/constants/.docs/after.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.docs/after.md:1]: We've recently [updated the spec](https://github.com/exercism/v3/issues/2293) (although we haven't updated the docs yet) which results in `after.md` documents being replaced with concept documents. Concretely, this file should be moved to `languages/go/concepts/constants/about.md`.

File: languages/go/exercises/concept/constants/.meta/example.go
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/.meta/example.go:41]: I don't see a mention of this type of definition in the `introduction.md`, is that right? If so, the student is unlikely to use this feature and as [per the docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#example-implementation-file):

> The implementation should only use language features introduced by the exercise or its prerequisites (and their prerequisites, and so on).

Is this feature introduced in one of the exercises prerequisites?

File: languages/go/exercises/concept/constants/constants.go
Comment (ErikSchierboom) [inline languages/go/exercises/concept/constants/constants.go:25]: Is this construct explained in the `introduction.md` file?

**Review by ErikSchierboom — APPROVED** [review]
I've left some comments. I've approved adhering to our optimistic-merging strategy. That said, I do think the story of this exercise will need a bit of work later. It might be good to create an issue for that.

**Comment by tehsphinx** [comment]
Merging would be great!

@ErikSchierboom: what are those checks that block this?

**Comment by ErikSchierboom** [comment]
@tehsphinx They are old checks. I think rebasing will solve this.

/rebase

**Comment by tehsphinx** [comment]
Seems it didn't help or didn't rebase. @ErikSchierboom, @jamessouth

**Comment by ErikSchierboom** [comment]
@tehsphinx I've gone and used my admin privileges. I would suggest to create a reference somewhere (e.g. in an issue) to fix the open comments.

Thanks for working on this @jamessouth!

**Comment by jamessouth** [comment]
Re Erik's open comments: the various concept exercises seem to be introducing their types individually instead of there being one overarching `types` lesson, whose issue is [on hold](https://github.com/exercism/v3/issues/836). `Strings` is a prerequisite of `constants`, but it does not include declaring custom types. It looks like we should either find a place prior to `constants` to teach custom types, or the use of the custom type feature in this lesson should be removed in accordance with the concept exercise design docs.

**Pull request merged**
